modules/mcc/include/opencv2/mcc/ccm.hpp

```diff
@@ -0,0 +1,522 @@
+
+    /** @brief set Distance
+    @param distance the type of color distance;\n
+                    default: @ref DISTANCE_CIE2000
+    */
+    CV_WRAP void setDistance(DISTANCE_TYPE distance);
+
+    /** @brief set Linear
+
+    Supported list:
+    @param linear_type the method of linearization;\n
+                       default: @ref LINEARIZATION_GAMMA
+    */
+    CV_WRAP void setLinear(LINEAR_TYPE linear_type);
+
+    /** @brief set Gamma
+
+    @note only valid when linear is set to "gamma";\n
+
+    @param gamma the gamma value of gamma correction;
+                 default: 2.2;\n
+    */
+    CV_WRAP void setLinearGamma(double gamma);
+
+    /** @brief set degree
+    @param deg the degree of linearization polynomial;\n
+        NOTICE: only valid when linear is set to
+        @ref LINEARIZATION_COLORPOLYFIT
+        @ref LINEARIZATION_GRAYPOLYFIT
+        @ref LINEARIZATION_COLORLOGPOLYFIT
+        @ref LINEARIZATION_GRAYLOGPOLYFIT
+               default: 3
+    */
+    CV_WRAP void setLinearDegree(int deg);
+
+    /** @brief set SaturatedThreshold
+    @param lower the lower threshold to determine saturation;\n
+            default: 0;
+    @param upper the upper threshold to determine saturation;
+        NOTICE: it is a tuple of [lower, upper];
+                The colors in the closed interval [lower, upper] are reserved to participate
+                in the calculation of the loss function and initialization parameters\n
+                 default: 0
+    */
+    CV_WRAP void setSaturatedThreshold(double lower, double upper);
+
+    /** @brief set WeightsList
+    @param weights_list the list of weight of each color;\n
+                        default: empty array
+    */
+    CV_WRAP void setWeightsList(Mat weights_list);
+
+    /** @brief set WeightCoeff
+    @param weights_coeff the exponent number of L* component of the reference color in CIE Lab color space;\n
+                         default: 0
+    */
+    CV_WRAP void setWeightCoeff(double weights_coeff);
+
+    /** @brief set InitialMethod
+    @param initial_method_type the method of calculating CCM initial value;\n
+    */
+    CV_WRAP void setInitialMethod(INITIAL_METHOD_TYPE initial_method_type);
+
+    /** @brief set MaxCount
+    @param max_count used in MinProblemSolver-DownhillSolver;\n
+        Terminal criteria to the algorithm;\n
+                     default: 5000;
+    */
+    CV_WRAP void setMaxCount(int max_count);
+
+    /** @brief set Epsilon
+    @param epsilon used in MinProblemSolver-DownhillSolver;\n
+        Terminal criteria to the algorithm;\n
+                   default: 1e-4;
+    */
+    CV_WRAP void setEpsilon(double epsilon);
+
+    /** @brief make color correction */
+    CV_WRAP void run();
+
+    CV_WRAP Mat getCCM() const;
+    CV_WRAP double getLoss() const;
+    CV_WRAP Mat get_src_rgbl() const;
+    CV_WRAP Mat get_dst_rgbl() const;
+    CV_WRAP Mat get_mask() const;
+    CV_WRAP Mat get_weights() const;
```

Comment:
Please follow OpenCV naming convention rules: https://github.com/opencv/opencv/wiki/Coding_Style_Guide#naming-conventions

`get_weights() => getWeights()`

---

Also it probably make sense to change return value `Mat => const Mat&`

Reply:
Thanks for reviewing the codes for us. We renamed functions under OpenCV rules.